lib/common/mobile/android/logcat-helper.ts

```diff
@@ -71,6 +75,41 @@ export class LogcatHelper implements Mobile.ILogcatHelper {
 					);
 				}
 			});
+
+			const appStartTrackingStream = await this.getAppStartTrackingLogcatStream(
+				deviceIdentifier,
+				options.appId
+			);
+
+			this.mapDevicesLoggingData[deviceIdentifier].appStartTrackingProcess =
+				appStartTrackingStream;
+
+			const rawLineStream = byline(appStartTrackingStream.stdout);
+			this.mapDevicesLoggingData[deviceIdentifier].rawLineStream =
+				rawLineStream;
+
+			rawLineStream.on("data", (lineBuffer: Buffer) => {
+				if (!this.mapDevicesLoggingData[deviceIdentifier]?.loggingProcess)
+					return;
+				const lines = (lineBuffer.toString() || "").split("\n");
+				for (let line of lines) {
+					// 09-11 17:50:26.311   598  1979 I ActivityTaskManager: START u0 {flg=0x10000000 cmp=org.nativescript.myApp/com.tns.NativeScriptActivity} from uid 2000
+					//                                                       ^^^^^                          ^^^^^^^^^^^^^^^^^^^^^^                                        ^^^^
+					// 																										   action                         appId                                                         pid
+
+					if (
+						// action
+						line.includes("START") &&
+						// appId
+						line.includes(options.appId) &&
+						// pid - only if it's not the current pid...
+						!line.includes(options.pid)
+					) {
+						this.forceStop(deviceIdentifier);
+						options.onAppRestarted?.();
```

Comment:
Note: not `break;`-ing here, since it's possible (though unlikely) that two subsequent START lines come in the same lineBuffer. Though this doesn't really matter - this stream is not noisy at all because we're pre-filtering with the adb logcat regex.